src/parsing.jl

```diff
@@ -257,12 +257,32 @@ end
 
 # ISO, YMD
 _DEFAULT_TYPE_MAP[:timestamptz] = ZonedDateTime
-const TIMESTAMPTZ_FORMATS = (
+const TIMESTAMPTZ_ZDT_FORMATS = (
     dateformat"y-m-d HH:MM:SSz",
     dateformat"y-m-d HH:MM:SS.sz",
     dateformat"y-m-d HH:MM:SS.ssz",
     dateformat"y-m-d HH:MM:SS.sssz",
 )
+const TIMESTAMPTZ_UTC_FORMATS = (
+    dateformat"y-m-d HH:MM:SS",
+    dateformat"y-m-d HH:MM:SS.s",
+    dateformat"y-m-d HH:MM:SS.ss",
+    dateformat"y-m-d HH:MM:SS.sss",
+)
+
+timestamptz_formats(::Type{ZonedDateTime}) = TIMESTAMPTZ_ZDT_FORMATS
+timestamptz_formats(::Type{UTCDateTime}) = TIMESTAMPTZ_UTC_FORMATS
+
+function _pqparse(::Type{T}, str::AbstractString) where T<:Union{UTCDateTime, ZonedDateTime}
+    formats = timestamptz_formats(T)
+    for fmt in formats[1:(end - 1)]
+        parsed = tryparse(T, str, fmt)
+        parsed !== nothing && return parsed
+    end
+
+    return parse(T, _trunc_seconds(str), formats[end])
+end
```

Comment:
With a more thorough look I realize this whole approach is unnecessary for UTCDateTime, and is why we chop off the `+00`. 

The only format necessary is `dateformat"y-m-d HH:MM:SS.s"`. It will parse all of these successfully:
```
"2022-01-01 01:02:30"
"2022-01-01 01:02:30.1"
"2022-01-01 01:02:30.12"
"2022-01-01 01:02:30.123"
```

The optional `_trunc_seconds` call is still necessary though.

Reply:
So LibPQ.jl is just working around TimeZones.jl not supporting the same parsing behaviour as Dates.jl?

Reply:
Not just TimeZones.jl; if you put any parsing element after the `.s` the automatic optional decimal feature goes away (this is why we couldn't have UTCDateTimes print a `Z` at the end and have that parse nicely).